src/config.ts

```diff
@@ -156,7 +156,7 @@ interface ParamConfig {
   textSteps?: number
   imageSteps?: number
   maxSteps?: number
-  restoreFaces?: boolean
+  restoreFaces?: 'CodeFormer' | 'GFPGAN' | boolean
```

Comment:
这个设计有一个小问题。当用户使用 sd-webui 并设置为 true 后，再次切换成 stable-horde，此时配置就从类型上不兼容了。反过来也有类似的问题。

这个问题并没有太好的解决方案。或许可以在两侧都加一个 transform 不过我不确定可行。但至少在实际代码中，可能需要给 stable-horde 传入的 true 一个 fallback。

Reply:
> 这个设计有一个小问题。当用户使用 sd-webui 并设置为 true 后，再次切换成 stable-horde，此时配置就从类型上不兼容了。反过来也有类似的问题。

Yeah, that is what I concerned about, either.
But I think it might be redundant to have `restoreFaces` and `restoreFacesModels` that can be covered by each other.
If you argue with the fallback method, I could change to this style.

> 这个问题并没有太好的解决方案。或许可以在两侧都加一个 transform 不过我不确定可行。

I think the `transform` is not a valid solution here.

> 但至少在实际代码中，可能需要给 stable-horde 传入的 true 一个 fallback。

Yep. So the `GFPGAN` is the primary face restoration model in the Stable Horde, I think it is better to treat `true` as `GFPGAN` if we were going to implement the fallback method.

Reply:
As sd-webui and stable horde both supports codeformers and GFPGAN as face restoration model, I think it might be better to produce a PR to webui side to make the face restoration configurable from API.
What do you think?

Reply:
> As sd-webui and stable horde both supports codeformers and GFPGAN as face restoration model, I think it might be better to produce a PR to webui side to make the face restoration configurable from API. What do you think?

Sounds great. You can also edit directly here and change the subject of this PR to "enhancing restore faces support".

Reply:
By the way, the Stable Horde called [CodeFormer](https://github.com/sczhou/CodeFormer) as "CodeFormer**s**", so we're also need to handle this weird differences. Lol.